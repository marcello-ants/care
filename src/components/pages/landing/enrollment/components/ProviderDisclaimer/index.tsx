// External Dependencies
import { Typography, Link } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

export default function ProviderDisclaimer() {
  // Styles
  const classes = useStyles();

  return (
    <Typography careVariant="disclaimer1" className={classes.disclaimer}>
      <span>
        You must be 18 years or older to use Care.com. By clicking &quot;Submit,&quot; you agree to
        our
      </span>{' '}
      <Link href="/terms-of-use-p1012.html" careVariant="disclaimer1">
        Terms of Use
      </Link>{' '}
      <span>and</span>{' '}
      <Link href="/privacy-policy-p1013.html" careVariant="disclaimer1">
        Privacy Policy
      </Link>
      <span>.</span>
    </Typography>
  );
}
