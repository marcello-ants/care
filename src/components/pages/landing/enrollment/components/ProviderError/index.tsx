// External Dependencies
import { Box } from '@material-ui/core';
import { Icon24UtilityWarning } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

function ProviderError() {
  // Styles
  const classes = useStyles();

  return (
    <Box className={classes.errorBanner}>
      <Box>
        <Icon24UtilityWarning color="#CF230C" />
      </Box>
      <Box>
        <Typography variant="h4">Whoops! Something went wrong</Typography>
        <Typography careVariant="body3">
          There was an error when we tried to submit your request. Please try again.
        </Typography>
      </Box>
    </Box>
  );
}

export default ProviderError;
