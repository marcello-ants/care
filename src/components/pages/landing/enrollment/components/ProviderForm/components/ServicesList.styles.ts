import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  empty: {
    '& > .MuiSelect-select': {
      color: theme.palette.care?.grey[700],
      fontWeight: 'normal',
    },
  },
  inputContainer: {
    width: '100%',
    '& > div': {
      padding: 0,
    },
  },
  error: {
    color: theme?.palette?.care?.red[800],
    marginLeft: theme.spacing(1.25),
    marginTop: theme.spacing(0.5),
  },
}));

export default useStyles;
