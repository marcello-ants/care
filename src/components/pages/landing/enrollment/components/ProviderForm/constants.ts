export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
