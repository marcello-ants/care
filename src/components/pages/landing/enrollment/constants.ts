export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
