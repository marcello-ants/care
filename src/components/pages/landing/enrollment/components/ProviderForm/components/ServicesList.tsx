/* eslint-disable import/prefer-default-export */
// External Dependencies
import { useState } from 'react';
import { Select, MenuItem, useMediaQuery } from '@material-ui/core';
import { useFormikContext } from 'formik';
import { theme } from '@care/material-ui-theme';
import { Typography } from '@care/react-component-lib';

// Internal Constants
import { PROVIDER_SERVICES_DESKTOP, PROVIDER_SERVICES_MOBILE } from '../constants';

// Styles
import useStyles from './ServicesList.styles';

function ServicesList() {
  // Styles
  const classes = useStyles();
  const isDesktop = useMediaQuery(theme.breakpoints.up('md'));

  const formik = useFormikContext<{ providerService: string }>();
  const listOfServices = isDesktop ? PROVIDER_SERVICES_DESKTOP : PROVIDER_SERVICES_MOBILE;

  // Component's State
  const [typeOfService, setTypeOfService] = useState('');

  // Handlers
  const onServiceValueChange = (value: string) => {
    formik.setFieldValue('providerService', value);
    setTypeOfService(value as string);
  };

  return (
    <>
      <Select
        name="selectJobType"
        className={
          !typeOfService
            ? `${classes.inputContainer} ${classes.empty}`
            : `${classes.inputContainer}`
        }
        value={typeOfService}
        displayEmpty
        onChange={(e) => {
          onServiceValueChange(e.target.value as string);
        }}>
        <MenuItem value="" disabled>
          Select one:
        </MenuItem>
        {listOfServices.map((service: any) => {
          return (
            <MenuItem value={service.value} key={service.value}>
              {service.label}
            </MenuItem>
          );
        })}
      </Select>
      {formik.errors.providerService && (
        <Typography careVariant="info1" className={classes.error}>
          Please select a job
        </Typography>
      )}
    </>
  );
}

export { ServicesList };
