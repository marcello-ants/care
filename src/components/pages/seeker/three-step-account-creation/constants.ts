import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
