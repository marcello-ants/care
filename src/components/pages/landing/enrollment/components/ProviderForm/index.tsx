// External Dependencies
import { useCallback } from 'react';
import getConfig from 'next/config';
import { FormikProvider } from 'formik';
import Button from '@material-ui/core/Button';
import { Box, Hidden } from '@material-ui/core';
import { Typography } from '@care/react-component-lib';

// Internal Dependencies
import FormikInlineTextField from '@/components/blocks/FormikInlineTextField';
import EmailField from '@/components/features/accountCreation/EmailField';
import PasswordField from '@/components/features/accountCreation/PasswordField';
import { HearAboutUsField } from '@/components/HearAboutUs';
import { useProviderEnrollmentForm } from './utils/formHandler';
import { ServicesList } from './components/ServicesList';
import ProviderError from '../ProviderError';
import ProviderDisclaimer from '../ProviderDisclaimer';

// Styles
import useStyles from './styles';

// Constants
const {
  publicRuntimeConfig: { CZEN_GENERAL_LOGIN },
} = getConfig();

function SeekerForm() {
  // Styles
  const classes = useStyles();

  // Formik
  const { formik, isSubmitError } = useProviderEnrollmentForm();

  // Handlers
  const handleFormSubmission = useCallback(() => {
    if (!formik.isValid || formik.isSubmitting) {
      return;
    }

    formik.handleSubmit();
  }, []);

  return (
    <>
      {isSubmitError && <ProviderError />}
      <FormikProvider value={formik}>
        <Typography variant="h2" className={classes.sectionTitle}>
          <span>What job are you most</span>{' '}
          <Hidden smUp>
            <br />
          </Hidden>{' '}
          <span>interested in?</span>
        </Typography>
        <ServicesList />
        <Box mt={5}>
          <Typography variant="h2" className={`${classes.sectionTitle}`}>
            <span>Create your account</span>
            <Typography careVariant="disclaimer1" className={classes.sectionTitleDisclaimer}>
              *required
            </Typography>
          </Typography>
        </Box>
        <Box className={classes.inputContainer}>
          <EmailField
            id="email"
            name="email"
            label="Email*"
            czenGeneralLoginUrl={CZEN_GENERAL_LOGIN}
            showErrorWhenTouched
          />
        </Box>
        <Box className={classes.inputsContainer}>
          <PasswordField
            id="password"
            name="password"
            label="Password*"
            defaultHelperText=""
            showErrorWhenTouched
          />
        </Box>
        <Box className={classes.inputContainer} mb={2}>
          <FormikInlineTextField id="providerZip" name="providerZip" label="Zip code*" />
        </Box>
        <Box className={classes.inputContainer}>
          <HearAboutUsField name="howDidYouHearAboutUs" />
        </Box>
        <Box mb={4}>
          <ProviderDisclaimer />
        </Box>
        <Button
          color="primary"
          variant="contained"
          size="large"
          className={classes.submitButton}
          disabled={formik.isSubmitting}
          onClick={handleFormSubmission}>
          Join now
        </Button>
      </FormikProvider>
      <Typography variant="body1" className={classes.companyInquiries}>
        <span>For company inquiries</span>{' '}
        <a href="https://www.care.com/business-p1369-q59964639.html">sign up here</a>
        <span>.</span>
      </Typography>
    </>
  );
}

export default SeekerForm;
