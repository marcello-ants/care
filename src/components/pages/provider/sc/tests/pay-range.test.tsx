import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { useRouter } from 'next/router';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { cloneDeep } from 'lodash-es';
import { GET_JOB_WAGES, SENIOR_CARE_PROVIDER_ATTRIBUTES_UPDATE } from '@/components/request/GQL';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { PROVIDER_ROUTES } from '@/constants';
import { SeniorCareProviderAttributesUpdateInput } from '@/__generated__/globalTypes';
import { seniorCareProviderAttributesUpdate } from '@/__generated__/seniorCareProviderAttributesUpdate';
import PayRange from '../pay-range';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

const initialStateClone = cloneDeep(initialAppState);
const initialStateOverrideWithZip = {
  ...initialStateClone,
  provider: {
    ...initialStateClone.provider,
    zipcode: '10004',
  },
};

const wageMock = {
  request: {
    query: GET_JOB_WAGES,
    variables: {
      zipcode: '10004',
      serviceType: 'SENIOR_CARE',
    },
  },
  result: {
    data: {
      getJobWages: {
        averages: {
          average: {
            amount: '20',
          },
          minimum: {
            amount: '17',
          },
          maximum: {
            amount: '26',
          },
        },
        legalMinimum: {
          amount: '15',
        },
      },
    },
  },
};
const wageMockError = {
  request: {
    query: GET_JOB_WAGES,
    variables: {
      zipcode: '10004',
      serviceType: 'SENIOR_CARE',
    },
  },
  error: new Error('aw shucks'),
};

const updateMock: MockedResponse<seniorCareProviderAttributesUpdate> = {
  request: {
    query: SENIOR_CARE_PROVIDER_ATTRIBUTES_UPDATE,
    variables: {
      input: {
        jobRate: {
          minimum: {
            amount: '17',
            currencyCode: 'USD',
          },
          maximum: {
            amount: '26',
            currencyCode: 'USD',
          },
        },
      } as SeniorCareProviderAttributesUpdateInput,
    },
  },
  result: {
    data: {
      seniorCareProviderAttributesUpdate: {
        __typename: 'SeniorCareProviderAttributesUpdateSuccess',
        dummy: 'hi',
      },
    },
  },
};

const updateErrorMock: MockedResponse<seniorCareProviderAttributesUpdate> = {
  request: {
    query: SENIOR_CARE_PROVIDER_ATTRIBUTES_UPDATE,
    variables: {
      input: {
        jobRate: {
          minimum: {
            amount: '17',
            currencyCode: 'USD',
          },
          maximum: {
            amount: '26',
            currencyCode: 'USD',
          },
        },
      } as SeniorCareProviderAttributesUpdateInput,
    },
  },
  error: new Error('An error occurred'),
};

let mockRouter = null;

async function renderResultAndWaitFinished(mocks: any) {
  const view = render(
    <MockedProvider mocks={mocks}>
      <AppStateProvider initialStateOverride={initialStateOverrideWithZip}>
        <PayRange />
      </AppStateProvider>
    </MockedProvider>,
    {}
  );
  await screen.findByRole('button', { name: 'Next' });

  return view;
}

beforeEach(async () => {
  mockRouter = {
    push: jest.fn(), // the component uses `router.push` only
    pathname: '',
  };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);
});

afterEach(() => {
  // cleanup on exiting
  mockRouter = null;
});

describe('Pay Range page', () => {
  it('matches snapshot', async () => {
    const { asFragment } = await renderResultAndWaitFinished([wageMock]);

    expect(asFragment()).toMatchSnapshot();
  });
  it('updates the slider value the from GQL call', async () => {
    await renderResultAndWaitFinished([wageMock]);

    const avgRange = screen.getByText('$17 - 26');
    expect(avgRange).not.toBeUndefined();
  });
  it('uses the slider defaults the GQL call error out', async () => {
    await renderResultAndWaitFinished([wageMockError]);

    // default wages
    const avgRange = screen.getByText('$14 - 22');
    expect(avgRange).not.toBeUndefined();
  });

  it('updates the slider value when moved', async () => {
    await renderResultAndWaitFinished([wageMock]);

    const minSlider = screen.queryAllByRole('slider')[0];
    fireEvent.keyDown(minSlider, { key: 'ArrowRight' });
    expect(screen.getByText('$18 - 26')).toBeVisible();
  });

  it('should navigate to the job-type page after successfully saving the selections', async () => {
    await renderResultAndWaitFinished([wageMock, updateMock]);
    fireEvent.click(screen.getByRole('button', { name: 'Next' }));
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_ROUTES.JOB_TYPE));
  });

  it('should display a message when an error is thrown', async () => {
    await renderResultAndWaitFinished([wageMock, updateErrorMock]);
    fireEvent.click(screen.getByRole('button', { name: 'Next' }));
    await waitFor(() =>
      expect(screen.getByText(/We weren't able to save your pay range/)).toBeVisible()
    );
  });

  it('should navigate to the job-type page after acknowledging the error message', async () => {
    await renderResultAndWaitFinished([wageMock, updateErrorMock]);
    fireEvent.click(screen.getByRole('button', { name: 'Next' }));
    await screen.findByText(/We weren't able to save your pay range/);
    fireEvent.click(screen.getByRole('button', { name: 'Got it' }));
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_ROUTES.JOB_TYPE));
  });
});
