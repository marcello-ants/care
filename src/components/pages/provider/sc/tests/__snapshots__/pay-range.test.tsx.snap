// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pay Range page matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h2
          class="MuiTypography-root makeStyles-header-2 MuiTypography-h2"
        >
          What is your preferred pay range?
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          class="MuiTypography-root makeStyles-careTypeContainer-3 MuiTypography-body2"
        >
          The average range in your area is $15 - $50.
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item"
        >
          <div
            class="MuiGrid-root makeStyles-bigLabeler-21 MuiGrid-item MuiGrid-grid-xs-12"
          >
            <span
              class="makeStyles-bigLabel-22 bigLabel"
            >
              $17 - 26
            </span>
            <span
              class="makeStyles-bigUnit-23 bigUnit"
            >
               / hr
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <span
              class="MuiSlider-root MuiSlider-colorPrimary makeStyles-slider-20"
            >
              <span
                class="MuiSlider-rail"
              />
              <span
                class="MuiSlider-track"
                style="left: 5.714285714285714%; width: 25.71428571428571%;"
              />
              <input
                type="hidden"
                value="17,26"
              />
              <span
                aria-orientation="horizontal"
                aria-valuemax="50"
                aria-valuemin="15"
                aria-valuenow="17"
                class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index="0"
                role="slider"
                style="left: 5.714285714285714%;"
                tabindex="0"
              />
              <span
                aria-orientation="horizontal"
                aria-valuemax="50"
                aria-valuemin="15"
                aria-valuenow="26"
                class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index="1"
                role="slider"
                style="left: 31.428571428571427%;"
                tabindex="0"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-nextButtonContainer-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </section>
</DocumentFragment>
`;
